loginCore/src/main/java/com/brandon/logincore/viewmodel/LoginViewModel.kt

```diff
@@ -0,0 +1,69 @@
+package com.brandon.logincore.viewmodel
+
+import android.net.ConnectivityManager
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.brandon.logincore.usecase.LoginUseCase
+import com.brandon.logincore.viewmodel.LoginViewModel.LoginUiState.*
+import com.brandon.utilities.LoginValidationState.*
+import com.brandon.utilities.isConnected
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.launch
+
+abstract class LoginViewModel : ViewModel() {
+
+    abstract val loginUseCase: LoginUseCase
+    abstract val connectivityManager: ConnectivityManager
+
+    private var _state: MutableStateFlow<LoginUiState> = MutableStateFlow(Idle())
```

Comment:
This should just live in the legacy one, no?

Reply:
adjusted to use only mutableStateOf in Compose